### PR TITLE
chore: bump dependencies

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

- update datafusion
- changed:
	- datafusion: rev xxx
	- arrow: `47.0`
	- parquet: `47.0`
	- tonic: `0.10`
	- sqlparser: `0.38`
	- prost: `0.12`
	- tonic-build: `0.10`
	- tonic-reflection: `0.10`
	- opentelemetry-proto: to my fork, rev `33841b38dda79b15f2024952be5f32533325ca02`. Wait on https://github.com/open-telemetry/opentelemetry-rust/pull/1310
	- substrait: `0.17`
- `trunc` is temporary removed from builtin py functions

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by evenyag — APPROVED** [review]

**Comment by waynexia** [comment]
it works on my machine.jpg

**Review by killme2008 — APPROVED** [review]

File: src/script/src/python/pyo3/builtins.rs [hunk 1]
```diff
@@ -90,7 +90,7 @@ pub(crate) fn greptime_builtins(_py: Python<'_>, m: &PyModule) -> PyResult<()> {
             floor,
             ceil,
             round,
-            trunc,
+            // trunc,
```
Comment (killme2008) [inline src/script/src/python/pyo3/builtins.rs:93]: Do we need to add a TODO?

File: src/common/time/src/timestamp.rs [hunk 2]
```diff
@@ -63,7 +63,7 @@ impl Timestamp {
             TimeUnit::Second => now.timestamp(),
             TimeUnit::Millisecond => now.timestamp_millis(),
             TimeUnit::Microsecond => now.timestamp_micros(),
-            TimeUnit::Nanosecond => now.timestamp_nanos(),
+            TimeUnit::Nanosecond => now.timestamp_nanos_opt().unwrap_or_default(),
```
Comment (evenyag) [inline src/common/time/src/timestamp.rs:66]: @v0y4g3r We updates chrono here.

File: tests/cases/standalone/common/basic.result [hunk 3]
```diff
@@ -4,7 +4,7 @@ CREATE TABLE system_metrics (
     cpu_util DOUBLE,
     memory_util DOUBLE,
     disk_util DOUBLE,
-    ts TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+    ts TIMESTAMP DEFAULT CURRENT_TIMESTAMP(),
```
Comment (evenyag) [inline tests/cases/standalone/common/basic.result:7]: Is `CURRENT_TIMESTAMP` an invalid syntax now? Maybe we need to update the document if it is an expected behavior.

Comment (waynexia) [reply tests/cases/standalone/common/basic.result:7]: y

File: src/script/src/python/pyo3/builtins.rs (hunk 1 above)
Comment (waynexia) [reply src/script/src/python/pyo3/builtins.rs:93]: Filed a issue https://github.com/GreptimeTeam/greptimedb/issues/2663

**Pull request merged**